tests/tclutils/popenRWE.cpp

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2009-2010 Bart Trojanowski <bart@jukie.net>
+ * Licensed under GPLv2, or later, at your choosing.
+ *
+ * bidirectional popen() call
+ *
+ * @param rwepipe - int array of size three
+ * @param exe - program to run
+ * @param argv - argument list
+ * @return pid or -1 on error
+ *
+ * The caller passes in an array of three integers (rwepipe), on successful
+ * execution it can then write to element 0 (stdin of exe), and read from
+ * element 1 (stdout) and 2 (stderr).
+ */
+
+#include "popenRWE.h"
+
+int popenRWE(int *rwepipe, const char *command) {
+  int in[2];
+  int out[2];
+  int err[2];
+  int pid;
+  int rc;
+
+  rc = pipe(in);
+  if (rc < 0) goto error_in;
+
+  rc = pipe(out);
+  if (rc < 0) goto error_out;
+
+  rc = pipe(err);
+  if (rc < 0) goto error_err;
+
+  pid = fork();
```

Comment:
Absolutely no Fork!
Please remove this component from the source tree. Forks are a terrible idea as they double the process memory size and and not portable.
Please revert back to the Pipe as I coded it.

Reply:
From the **popen()** documentation :
The popen() function opens a process by creating a pipe, forking,
       and invoking the shell.

Reply:
this approach is the same as it was before but with error handling